### DISCUSSION TITLE
optimize WITH clauses that are referenced more than once

### DISCUSSION THREAD

**Discussion by laidongxu** [trinodb/trino#18422] [Q&A]
By testing and reviewing the trino execution plan, when a query with clauses is used multiple times, with clauses will be executed multiple times
Why not optimize with clauses within Trino ？

**Comment by hashhar** [comment]
See https://github.com/trinodb/trino/issues/5878 and https://github.com/trinodb/trino/issues/5880. Follow them to keep updated.
